src/Symfony/Bundle/FrameworkBundle/Resources/config/console.xml

```diff
@@ -73,9 +73,8 @@
             <argument /> <!-- Message bus locator -->
             <argument type="service" id="messenger.receiver_locator" />
             <argument type="service" id="logger" on-invalid="null" />
-            <argument type="collection" /> <!-- Receiver names -->
-            <argument type="collection" /> <!-- Message bus names -->
-
+            <argument>%messenger.receiver_names%</argument>
```

Comment:
the purpose of this argument is only to improve DX by suggesting alternatives when you try to execute the command with an undefined receiver as argument. 
Overriding this parameter to add more service names won't make those services available to the command since the locator remains unchanged. It would just give wrong suggestions in case an undefined receiver is passed. 
Note that If you are truly decorating receivers, your decorators should be available to the command under the name of the decorated receiver as decorating means replacing (dependents always get the decorator injected).

Reply:
@chalasr I know that it's only for DX, but it's something that a custom consumer command should use since it gives the ability for the command to print out a list of valid receivers to input.

This isn't for decorating receivers, but it's for allowing custom consumer commands that might bind their own decorators at runtime (like the current consume command does). And when creating a new receiver, if you want to have access to the receiver names for DX, you'll need to make compiler pass to find all tagged services and get the aliases to make the names.

I'm willing to explore other options as well, but this seemed like the most straight forward.

Reply:
@ragboyjr Could you attach some code illustrating what this PR solves? I'm not sure to fully understand

Reply:
Sure @chalasr.

Here's my specific use case:

I've built a custom receiver enhancers that will actually auto scale processes to handle the requests coming in. It uses pcntl and it measures the rate of messages coming in and if there are more messages being put in vs taken out, it adds another worker, when the workload is finished it will start scaling the procs down.

There is currently no way when using the default `messenger:consume-messages` command to add this decorator in the command, instead, I need to make my own command like `app:consume-messages` which properly adds the autoscaling decorator. This command is basically everything like messenger:consume-messages command is, but the code for decorating is different.

So, to get the receiver names to show up for my `app:consume-messages` command, I need to create a CompilerPass which does the same exact code the MessengerCompilerPass does for building up and passing in the receiver_names to the commands arguments.

This PR would make it so that any custom commands could easily just use those receiver names and not need to create a compiler pass every time you make your own consume messages command.